src/nvim/api/extmark.c

```diff
@@ -1211,3 +1222,73 @@ String nvim__buf_debug_extmarks(Buffer buffer, Boolean keys, Boolean dot, Error
 
   return mt_inspect(buf->b_marktree, keys, dot);
 }
+
+/// Adds the namespace scope to the window.
+///
+/// @param window Window handle, or 0 for current window
+/// @param ns_id the namespace to add
+/// @return true if the namespace was added, else false
+Boolean nvim_win_add_ns(Window window, Integer ns_id, Error *err)
+  FUNC_API_SINCE(12)
+{
+  win_T *win = find_window_by_handle(window, err);
+  if (!win) {
+    return false;
+  }
+
+  VALIDATE_INT(ns_initialized((uint32_t)ns_id), "ns_id", ns_id, {
+    return false;
+  });
+
+  set_put_uint32_t(&win->w_ns_set, (uint32_t)ns_id, NULL);
```

Comment:
`set_put(uint32_t, &win->w_ns_set, (uint32_t)ns_id, NULL)` (and similar for set_put/set_has)